### PR TITLE
coeff_dir in GUI is now relative to config_dir

### LINKED ISSUE #19: Use relative paths for filter coeff files
@JWahle is the realtive path against the active config not working something temporary ?

Before it worked well with relative paths against the config.yml for the plotting.
This will make it less portable for users (they have to know where the the files will be on the specific target).
If it keeps this way I need to patch in moOde all existing or new user uploaded config files to use abs path.

_Originally posted by @bitkeeper in https://github.com/HEnquist/camillagui-backend/issues/18#issuecomment-784221869_

### PR DESCRIPTION
**Pull request by JWahle**
- filter paths in the GUI and stored in the backend are now relative
- on config/active_config load filter paths are converted to relative
- for filter/pipeline_step plot filter paths are converted to absolute

I also removed unused filter and pipeline step routes and corresponding functions.

Fixes #19 

@HEnquist please have a look.
Loading of configs with absolute and relative convolution filter files should work.
Plotting of filters and pipeline steps should work for files with absolute and relative filter files.

### CODE REVIEW

File: routes.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+
 from views import (
     get_param,
     get_list_param,
     set_param,
     eval_filter_values,
     eval_filterstep_values,
-    eval_filter_svg,
-    eval_filterstep_svg,
     eval_pipeline_svg,
```
Comment (HEnquist) [inline routes.py:9]: Very welcome cleanup! Let's get rid of `eval_pipeline_svg` too.

File: views.py [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+    zip_response, zip_of_files, get_yaml_as_json, set_as_active_config, get_active_config, save_config,
+    new_config_with_absolute_filter_paths, coeff_dir_relative_to_config_dir,
+    replace_relative_filter_path_with_absolute_paths, new_config_with_relative_filter_paths
 )
 from offline import cdsp_or_backup_cdsp, set_cdsp_config_or_validate_with_backup_cdsp
 from settings import gui_config_path
 
 try:
-    from camilladsp_plot import plot_pipeline, plot_filter, plot_filterstep
+    from camilladsp_plot import plot_pipeline, eval_filter, eval_filterstep
```
Comment (HEnquist) [inline views.py:16]: questionable if we should keep this. The plotting lib can do all the filter eval without using any extra library (like the big numpy or pyplot) so there shouldn't be anything preventing the backend from requiring it.

**Comment by HEnquist** [comment]
I have looked through this now (but not tried running it yet) and I think it looks good. Seems like a good way to handle the relative/absolute problem.

File: views.py (hunk 2 above)
Comment (JWahle) [reply views.py:16]: Cool!
Shall I remove the numpy and pyplot libraries from the README.md?

Comment (HEnquist) [reply views.py]: If you don't mind, then yes please go ahead and remove everything about pyplot and numpy!

**Comment by JWahle** [comment]
@HEnquist I updated the `README.md`. Please merge when you are ready.

**Comment by JWahle** [comment]
@HEnquist do you want to do some testing/review, or shall I just merge?

**Comment by HEnquist** [comment]
Finally got time to test this, and it's working just fine on both linux and windows. Merging!

**Pull request merged**